src/main/cms/cms.c

```diff
@@ -161,9 +161,10 @@ bool cmsDisplayPortSelect(displayPort_t *instance)
 //   13 cols x 9 rows, top row printed as a Bold Heading
 //   Needs the "smallScreen" adaptions
 
-#define CMS_MAX_ROWS 16
+#define CMS_MAX_ROWS 31
```

Comment:
30 + 1

Reply:
2^5 - 1

Reply:
I'm not quite sure what happened to my comment on this, I submitted it but it appears to have been lost, I'll re-state my comment which was:

"Shouldn't this value come from the OSD subsystem as it ultimately the OSD subsystem that is responsible for deciding how many rows there are".

Reply:
It’s doesn’t necessarily follow that CMS runs on the OSD. We could have used `(1 << OSD_POSITION_BITS) - 1` I guess, but didn’t because of the lack of a direct dependency on OSD.

Reply:
Ahh yes, regarding a specific OSD impl dependency.  

However at the time it is invoked at runtime it depends on the displayport API to do all the character drawing, so it should be at least use the displayport API to get the current displayport target's maximum rows and work accordingly.